# <font color=blue>Assignments for "Data Cleaning - Variable Types"</font>

**(1)** Determine the types of the below-given variables:
- Age
- Salary
- Income
- Customer type
- Stock price 

In [ ]:
"""
Age : Discrete Variable
Salary : Can be regarded as a Continuous Variable, can be regarded as a Discrete Variable, too.
Income : Same as Salary
Customer Type : Nominal & Ordinal
Stock Price : Discrete Variable
"""

SyntaxError: ignored

**(2)** Using the `movie` data, we have studied in this lesson. Complete the following tasks:
    * Select a variable from the `movie` dataset and discuss the variable type of this variable
    * Categorize a variable of your choice and assign a name to it
    * Count the number of rows for each category

In [ ]:
import pandas as pd
import numpy as np
movie = pd.read_csv("movie_metadata.csv")
movie.info()

In [ ]:
movie.imdb_score

#as one can see, imdb_scores are discrete *variables*

In [ ]:
def greater_than_7(imdb_score):
  if imdb_score > 7.0:
    return 1
  else:
    return 2

In [ ]:
movie['imdb_scores_greater_than_7'] = movie['imdb_score'].apply(greater_than_7)
display(movie.head())
movie.groupby("imdb_scores_greater_than_7")["imdb_score"].count()

color      director_name  ...  movie_facebook_likes  imdb_scores_greater_than_7
0  Color      James Cameron  ...                 33000                           1
1  Color     Gore Verbinski  ...                     0                           1
2  Color         Sam Mendes  ...                 85000                           2
3  Color  Christopher Nolan  ...                164000                           1
4    NaN        Doug Walker  ...                     0                           1

[5 rows x 29 columns]

imdb_scores_greater_than_7
1    1594
2    3449
Name: imdb_score, dtype: int64

# Hence there are 1594 movies that have imdb score greater than 7, whereas 3449 movies have less than 7.

[link text](https://)**(3)** Is it possible to convert categorical variable into numerical one? If so, how can you make that?

In [ ]:
# We could assign the colors to diverse numbers and thus we'd convert the categorical values into numerical ones.

**(4)** While continuous data is generally preferred over discrete data, please indicate circumstances where discrete is the preferred data type although continuous data is available for the same characteristic.

In [ ]:
"""
As for in the  ages of people, the data can be written in 19.912318239 years old, as in the limit approach in Mathematics.
But we'd like to deal with grouped data since it is easier to deal with than the other version of data
"""

**(5)** While continuous data is measured and attribute data is counted, there is sometimes confusion if some specific dataset should be considered continuous or attribute. Provide some examples of confusing datasets and your inference. 

In [ ]:
# Well, the data of the human hair color may be considered as categorical data, but for some cases we can 'see' them as if they are numerical data.